From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Wayne Chang <waynec@nvidia.com>
Date: Tue, 3 Sep 2024 09:04:19 +0000
Subject: [PATCH 152/243] NVIDIA: SAUCE: usb: gadget: tegra-xudc: Increase
 tPortConfiguration timeout

The default tPortConfiguration timeout is too small.
Enlarge the timeout configuration to ensure better reliability.

Bug 4786323

Change-Id: I56e828d19e806a170e13d72f3339f24c9ea5f78f
Signed-off-by: Wayne Chang <waynec@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3250528
Reviewed-by: Henry Lin <henryl@nvidia.com>
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
Reviewed-by: WK Tsai <wtsai@nvidia.com>
Reviewed-by: svcacv <svcacv@nvidia.com>
Reviewed-by: HaoTien Hsu <haotienh@nvidia.com>
(cherry picked from commit 6b804b9fb4aee05358f0023ecb5156c481d9b74d)

diff --git a/drivers/usb/gadget/udc/tegra-xudc.c b/drivers/usb/gadget/udc/tegra-xudc.c
index e425d2c0001a..522a8b89b818 100644
--- a/drivers/usb/gadget/udc/tegra-xudc.c
+++ b/drivers/usb/gadget/udc/tegra-xudc.c
@@ -146,10 +146,6 @@
 #define SSPX_CORE_CNT0 0x610
 #define  SSPX_CORE_CNT0_PING_TBURST_MASK GENMASK(7, 0)
 #define  SSPX_CORE_CNT0_PING_TBURST(x) ((x) & SSPX_CORE_CNT0_PING_TBURST_MASK)
-#define SSPX_CORE_CNT30 0x688
-#define  SSPX_CORE_CNT30_LMPITP_TIMER_MASK GENMASK(19, 0)
-#define  SSPX_CORE_CNT30_LMPITP_TIMER(x) ((x) & \
-					SSPX_CORE_CNT30_LMPITP_TIMER_MASK)
 #define  SSPX_CORE_CNT32_POLL_TBURST_MAX_MASK GENMASK(7, 0)
 #define  SSPX_CORE_CNT32_POLL_TBURST_MAX(x) ((x) & \
 					SSPX_CORE_CNT32_POLL_TBURST_MAX_MASK)
@@ -3521,11 +3517,18 @@ static void tegra186_xudc_device_params_init(struct tegra_xudc *xudc)
 	val |= SSPX_CORE_CNT0_PING_TBURST(0xa);
 	xudc_writel(xudc, val, SSPX_CORE_CNT0);
 
+#define SSPX_CORE_CNT30 0x688
+#define  SSPX_CORE_CNT30_LMPITP_TIMER_MASK GENMASK(19, 0)
+#define  SSPX_CORE_CNT30_LMPITP_TIMER(x) ((x) & \
+					SSPX_CORE_CNT30_LMPITP_TIMER_MASK)
 	/* Default tPortConfiguration timeout is too small. */
 	val = xudc_readl(xudc, SSPX_CORE_CNT30);
 	val &= ~(SSPX_CORE_CNT30_LMPITP_TIMER_MASK);
 	val |= SSPX_CORE_CNT30_LMPITP_TIMER(0x978);
 	xudc_writel(xudc, val, SSPX_CORE_CNT30);
+#undef SSPX_CORE_CNT30
+#undef SSPX_CORE_CNT30_LMPITP_TIMER_MASK
+#undef SSPX_CORE_CNT30_LMPITP_TIMER
 
 	/*
 	 * Compliance suite appears to be violating polling
@@ -3566,6 +3569,19 @@ static void tegra264_xudc_device_params_init(struct tegra_xudc *xudc)
 	val |= SSPX_CORE_CNT27_PING_LFPS_TRPT(0x7a1200);
 	xudc_writel(xudc, val, SSPX_CORE_CNT27);
 
+#define SSPX_CORE_CNT30 0x4088
+#define  SSPX_CORE_CNT30_LMPITP_TIMER_MASK GENMASK(11, 0)
+#define  SSPX_CORE_CNT30_LMPITP_TIMER(x) ((x) & \
+					SSPX_CORE_CNT30_LMPITP_TIMER_MASK)
+       /* Default tPortConfiguration timeout is too small. */
+	val = xudc_readl(xudc, SSPX_CORE_CNT30);
+	val &= ~(SSPX_CORE_CNT30_LMPITP_TIMER_MASK);
+	val |= SSPX_CORE_CNT30_LMPITP_TIMER(0xc9);;
+	xudc_writel(xudc, val, SSPX_CORE_CNT30);
+#undef SSPX_CORE_CNT30
+#undef SSPX_CORE_CNT30_LMPITP_TIMER_MASK
+#undef SSPX_CORE_CNT30_LMPITP_TIMER
+
 #define SSPX_CORE_CNT32 0x4090
 	/*
 	 * Compliance suite appears to be violating polling
-- 
2.43.0

